packages/@react-spectrum/searchwithin/src/SearchWithin.tsx

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright 2020 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+import {classNames, SlotProvider, useFocusableRef, useResizeObserver, useStyleProps} from '@react-spectrum/utils';
+import {DOMProps, FocusableRef, SpectrumLabelableProps, StyleProps} from '@react-types/shared';
+import {Field} from '@react-spectrum/label';
+import React, {ReactNode, useCallback, useLayoutEffect, useRef, useState} from 'react';
+import styles from '@adobe/spectrum-css-temp/components/searchwithin/vars.css';
+import {useLabel} from '@react-aria/label';
+import {useProvider, useProviderProps} from '@react-spectrum/provider';
+
+export interface SpectrumSearchWithinProps extends SpectrumLabelableProps, DOMProps, StyleProps {
+  children: ReactNode,
+  /** Whether the children should be disabled. Propagated to both children. */
+  isDisabled?: boolean
+}
+
+function SearchWithin(props: SpectrumSearchWithinProps, ref: FocusableRef<HTMLElement>) {
+  props = useProviderProps(props);
+  let {styleProps} = useStyleProps(props);
+  let {labelProps} = useLabel(props);
+  let {
+    children,
+    isDisabled,
+    isRequired
+  } = props;
+
+  let inputRef = useRef<HTMLInputElement | HTMLTextAreaElement>();
+  let domRef = useFocusableRef(ref, inputRef);
+
+  // Measure the width of the field to inform the width of the menu.
+  let [menuWidth, setMenuWidth] = useState(null);
+  let {scale} = useProvider();
+
+  let onResize = useCallback(() => {
+    if (domRef.current) {
+      let width = domRef.current.offsetWidth;
+      setMenuWidth(width);
+    }
+  }, [domRef]);
+
+  useResizeObserver({
+    ref: domRef,
+    onResize: onResize
+  });
+
+  useLayoutEffect(onResize, [scale, onResize]);
+
+  let defaultSlotValues = {isDisabled, isRequired, label: undefined, isQuiet: false, 'aria-labelledby': labelProps.id};
+  let searchFieldClassName = classNames(styles, 'spectrum-SearchWithin-input');
+  let pickerClassName = classNames(styles, 'spectrum-SearchWithin-picker');
+  let slots = {
+    searchfield: {UNSAFE_className: searchFieldClassName, ...defaultSlotValues},
+    picker: {UNSAFE_className: pickerClassName, menuWidth, align: 'end', ...defaultSlotValues}
```

Comment:
why align: 'end'?

Reply:
Otherwise, the menu is positioned as shown:

<img width="396" alt="Screen Shot 2021-06-29 at 2 38 13 PM" src="https://user-images.githubusercontent.com/8961049/123857310-bc1d1680-d8e7-11eb-9a62-2c895edfb542.png">

Let me know if there is a better method.

Reply:
o, yep, forgot that applied to the picker's menu